### ISSUE TITLE
Add "Related file" knowledge

### ISSUE THREAD

**Issue by mterwoord** [JetBrains/resharper-devguide#43]
I want to add support for my framework. For that, I need a way to "feed" ReSharper with "Related Files" rules, knowledge, info, etc. Any way to do that?

**Comment by citizenmatt** [comment]
Sorry, missed this comment. I'm not sure what you're asking with this question, it needs more detail. Also, if it's a question about working with the SDK, rather than an issue with the documentation, it would probably be better to post to StackOverflow (with the resharper tag) or to the [Google Group](https://groups.google.com/forum/#!forum/resharper-plugins).

**Comment by mterwoord** [comment]
Not sure if my "problem" is with the SDK or the docs, but I'd expect info on giving resharper the info needed in the docs.

**Comment by citizenmatt** [comment]
Yep, I agree. Unfortunately, writing the docs is a lengthy process, as they need to cover the topic more thoroughly than just addressing a particular question. Raising an issue here will highlight that the docs need updating, and that work will begin, but it won't necessarily answer your particular question in a timely fashion. Something like StackOverflow or the Google group will also get the question out to the wider community, and also in a place where it's likely to be found again when searching. This issue tracker isn't the best place to come to search for answers on implementation, more for tracking what parts of the docs need to be updated. I'd expect to add an issue here in response to a question on SO.

What info are you trying to give to ReSharper?

**Comment by mterwoord** [comment]
I have a library which uses certain conventions. 
I want to use Navigate to Related Files on model class, and then my resharper plugin should be able to figure out which interface implementations to show..
I'm just assuming the SDK is able to let me do this.

I'll work on some SO questions then..

**Comment by citizenmatt** [comment]
You might want to take a look at these SO answers:

http://stackoverflow.com/questions/30918419/create-a-generic-type-to-find-implementations-of-in-a-resharper-plugin

and

http://stackoverflow.com/questions/31044308/how-do-i-create-a-itypeelement-with-represents-a-closed-generic-type-in-a-resh

**Comment by mterwoord** [comment]
Awesome. I don't have time now, but those 2 definitely seem to provide me with some good info! Thanks!
